astropy/time/formats.py

```diff
@@ -613,6 +616,10 @@ def _check_val_type(self, val1, val2):
     def set_jds(self, val1, val2):
         """Convert datetime object contained in val1 to jd1, jd2"""
         # Iterate through the datetime objects, getting year, month, etc.
+        if self._check_val_empty(val1):
+            self.jd1 = self.jd2 = np.array([])
+            return
+
         iterator = np.nditer([val1, None, None, None, None, None, None],
```

Comment:
I think for many cases, one doesn't have to special-case the empty shape, but one should just pass `'zerosize_ok'` in the flags of `nditer` below (see https://github.com/astropy/astropy/issues/7866#issuecomment-427428243)